Client/src/Unified/Services/Submitter/Service.cs

```diff
@@ -94,22 +99,71 @@ public class Service : AbstractClientService, ISubmitterService
     : base(properties,
            loggerFactory)
   {
+    var timeOutSending = properties.TimeTriggerBuffer ?? TimeSpan.FromSeconds(1);
+
+    var maxTasksPerBuffer = properties.MaxTasksPerBuffer;
+
+    semaphoreSlim_ = new SemaphoreSlim(properties.MaxConcurrentBuffer * maxTasksPerBuffer);
+
+    queue_ = Channel.CreateUnbounded<string>(new UnboundedChannelOptions());
+
     SessionServiceFactory = new SessionServiceFactory(LoggerFactory);
 
     SessionService = SessionServiceFactory.CreateSession(properties);
 
     ProtoSerializer = new ProtoSerializer();
 
     CancellationResultTaskSource = new CancellationTokenSource();
+    CancellationQueueTaskSource  = new CancellationTokenSource();
 
     HandlerResponse = Task.Run(ResultTask,
                                CancellationResultTaskSource.Token);
 
     Logger = LoggerFactory?.CreateLogger<Service>();
     Logger?.BeginPropertyScope(("SessionId", SessionService.SessionId),
                                ("Class", "Service"));
+
+    BufferSubmit = new BatchUntilInactiveBlock<BlockRequest>(maxTasksPerBuffer,
+                                                             timeOutSending,
+                                                             new ExecutionDataflowBlockOptions
+                                                             {
+                                                               BoundedCapacity        = properties.MaxParallelChannel,
```

Comment:
Yes MaxParallelChannel here. 

- MaxTasksPerBuffer represents the number of BlockRequest in BatchBlock

- MaxParallelChannel : It's for transformBlock or ActionBlock which have at this stage of dataflow a BlockRequest[]. it will authorize and limit the capacity to the max number batchblock per channel